### PR TITLE
BUG: Adding empty dataframes should result in empty blocks #10181

### LINKED ISSUE #10181: Empty dataframe not equal to another empty dataframe in 0.16.1
I searched the issues list for "empty dataframe"  but did not find anything matching. 

In pandas 0.16.1, the result of adding 2 empty dataframes  is not equal to another empty dataframe. 

This code, in pandas 0.15.2, does what is expected: 

```python
import pandas as pd

pd.show_versions()

df1 = pd.DataFrame()

df2 = pd.DataFrame()

assert df1.equals(df1)

assert df2.equals(df1)

assert (df1+df1).equals(df2)
```

It fails in 0.16.1.

0.15.2 output:
```python

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.7.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 26 Stepping 5, GenuineIntel
byteorder: little
LC_ALL: None
LANG: en_GB

pandas: 0.15.2
nose: 1.3.6
Cython: 0.21.1
numpy: 1.9.2
scipy: 0.15.1
statsmodels: None
IPython: 3.1.0
sphinx: 1.3.1
patsy: 0.3.0
dateutil: 2.4.2
pytz: 2015.4
bottleneck: 0.8.0
tables: 3.1.1
numexpr: 2.3.1
matplotlib: 1.4.0
openpyxl: 1.8.5
xlrd: 0.9.3
xlwt: 0.7.5
xlsxwriter: 0.7.2
lxml: 3.4.0
bs4: 4.3.2
html5lib: 0.999
httplib2: None
apiclient: None
rpy2: None
sqlalchemy: 0.9.7
pymysql: None
psycopg2: None
```

0.16.1 output:
```python
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.9.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 26 Stepping 5, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.16.1
nose: 1.3.6
Cython: 0.22
numpy: 1.9.2
scipy: 0.14.0
statsmodels: 0.6.1
IPython: 3.1.0
sphinx: 1.3.1
patsy: 0.3.0
dateutil: 2.4.2
pytz: 2015.4
bottleneck: 0.8.0
tables: 3.1.1
numexpr: 2.3.1
matplotlib: 1.4.3
openpyxl: None
xlrd: 0.9.3
xlwt: None
xlsxwriter: 0.7.2
lxml: None
bs4: 4.3.2
html5lib: 0.999
httplib2: None
apiclient: None
sqlalchemy: 1.0.4
pymysql: None
psycopg2: None
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
c:\test_empty_add_0.16.1.py in <module>()
     11 assert df2.equals(df1)
     12
---> 13 assert (df1+df1).equals(df2)

AssertionError:
```

### PR DESCRIPTION
**Pull request by rekcahpassyla**
Fixes #10181 

I added a couple of lines to the dataframe test `test_operators` to assert that  adding two empty dataframes returns a dataframe that is equal (both using `assert_frame_equal` as well as `.equals`) to an empty dataframe.

I couldn't get vbench to install (I am running on windows) - the error was:

```
c:\dev\code\opensource\pandas-rekcahpassyla>pip install git+https://github.com/pydata/vbench
You are using pip version 6.0.8, however version 6.1.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Collecting git+https://github.com/pydata/vbench
  Cloning https://github.com/pydata/vbench to c:\users\redacted\appdata\local\temp\pip-ruiahg-build
fatal: unable to connect to github.com:
github.com[0: 192.30.252.129]: errno=No error

Clone of 'git://github.com/yarikoptic/vbenchtest.git' into submodule path 'vbench/tests/vbenchtest/vbenchtest' failed
  Complete output from command C:\dev\bin\git\cmd\git.exe submodule update --init --recursive -q:

  ----------------------------------------
  Command "C:\dev\bin\git\cmd\git.exe submodule update --init --recursive -q" failed with error code 1 in c:\users\redacted\appdata\local\temp\pip-ruiahg-build
```

The test suite failed for me in master with 1 failure. The same failure, but no others, were observed when running in my branch.

### CODE REVIEW

File: pandas/tests/test_frame.py [hunk 1]
```diff
@@ -5129,6 +5129,11 @@ def test_operators(self):
         df = DataFrame({'a': ['a', None, 'b']})
         assert_frame_equal(df + df, DataFrame({'a': ['aa', np.nan, 'bb']}))
 
+        df = DataFrame()
+        self.assertTrue((df + df).equals(DataFrame()))
```
Comment (jreback) [inline pandas/tests/test_frame.py]: pls add the issue number as a comment.

I would also add a couple of other test, things like

```
df = DataFrame(columns=['A'])
self.assertTrue((df+df).equals(df)
```
and add for an index, but no columns

**Comment by rekcahpassyla** [comment]
Added more tests and comment as suggested

**Comment by jreback** [comment]
@rekcahpassyla ok this looks reasonable. pls add a release note to whatsnew/v0.17.0 (you can put in other API changes section).

pls squash as well.

**Comment by rekcahpassyla** [comment]
Added release note and squashed.

**Comment by rekcahpassyla** [comment]
I note the build failure, will have a look.

**Comment by rekcahpassyla** [comment]
I have verified that my latest commit passes the `pandas.stats` tests (the cause of the build failure). However I haven't squashed it, as I don't know if it would be considered a separate bug (searching the issues list didn't reveal anything). If it is a separate bug, I'm happy to open another issue and cherrypick this commit into another PR.

**Comment by rekcahpassyla** [comment]
Raised  https://github.com/pydata/pandas/issues/10195 and proposed https://github.com/pydata/pandas/pull/10196 to fix the issue causing https://travis-ci.org/pydata/pandas/builds/63544612 to fail.

**Comment by jreback** [comment]
pls rebase this (and move the release note to 0.16.2)

**Comment by jreback** [comment]
I think these are some old travis failures. can you rebase on master. ping when green.

**Comment by rekcahpassyla** [comment]
@jreback - It is green

**Comment by jreback** [comment]
thanks!

**Pull request merged**